### PR TITLE
Ex 01

### PR DESCRIPTION
**Pull request by vsnupoudel**
Creating pull request for exercise 01

### CODE REVIEW

File: src/bishnu_poudel_ex/ex01/comp_to_loop.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+# Checking the output---
+# a=squares_by_comp(10)
+# print(a)
+# print("-----------------------------------------------------------------------------")
+# print("-----------------------------------------------------------------------------")
+# print("-----------------------------------------------------------------------------")
+# b=squares_by_loop(10)
+# print(b)
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/comp_to_loop.py]: This is not TODO, or a proposal for a solution. Please remove this.

File: src/bishnu_poudel_ex/ex01/deck_by_comp.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    deck = []
+    for suit in SUITS:
+        for val in VALUES:
+            deck.append((suit, val))
+    return deck
+
+
+def deck_comp():
+    deck = []
+    [deck.append((suit, val)) for suit in SUITS for val in VALUES]
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/deck_by_comp.py]: After this line, in the memory, you will have:
- `deck` with data as required
- A unassigned list of decks (and unused)

List comprehension is for creating a variable that require loop, not a short-hand for loop.
In this case `return [(suit, val) for suit in SUITS for val in VALUES]` is enough for the answer.

File: src/bishnu_poudel_ex/ex01/deck_by_comp.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+# Checking the output---
+# a=deck_loop()
+# print(a)
+# print("-----------------------------------------------------------------------------")
+# print("-----------------------------------------------------------------------------")
+# print("-----------------------------------------------------------------------------")
+# b=deck_comp()
+# print(b)
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/deck_by_comp.py]: Same reason with the above comment. Remove these.

File: src/bishnu_poudel_ex/ex01/letter_counts.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+"""Count distinct characters in a text- for Assignment 1"""
+__author__ = "Bishnu Poudel"
+__email__ = "bishnu.poudel@nmbu.no"
+
+
+def letter_freq(txt):
+    new_dict = dict()
+    for i in range(len(txt)):
+        if txt[i] in new_dict.keys():
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/letter_counts.py]: This is too complicated to get a character in a string.
You can do this:
```
for char in txt:
   if char in new_dict.keys():
```
Also, the exercise requires you to ignore the case of the letter, which means `H` and `h` should be counted in the same letter category. Thus, you may need to use `lower()` function here.

File: src/bishnu_poudel_ex/ex01/letter_counts.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+            # print("TRUE")
+            new_dict[txt[i]] = new_dict[txt[i]] + 1
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/letter_counts.py]: You can use short-hand here:
`new_dict[char] += 1`

File: src/bishnu_poudel_ex/ex01/tidy_code.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+"""Cleaning the code for Exercises fro Assignment 1"""
+__author__ = "Bishnu Poudel"
+__email__ = "bishnu.poudel@nmbu.no"
+
+from random import randint as a
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/tidy_code.py]: Renaming to `a` makes it hard to understand what's the function is doing.
In this case,
`from random import randint` is enough.

File: src/bishnu_poudel_ex/ex01/tidy_code.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+
+# main code
+if __name__ == '__main__':
+    statenow = False
+    attempts_to_guess = 3
+    rand_int = rand_int_between_1_12()
+    # print('rand int is {} '.format(rand_int))
+    while not statenow and attempts_to_guess > 0:
+        k = input_guess()
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/tidy_code.py]: Can you think of a better name for `k`?

File: src/bishnu_poudel_ex/ex01/tidy_code.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
+"""Cleaning the code for Exercises fro Assignment 1"""
+__author__ = "Bishnu Poudel"
+__email__ = "bishnu.poudel@nmbu.no"
+
+from random import randint as a
+
+
+# compare two numbers
+def compare(f, g):
+    return f == g
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/tidy_code.py]: Can you think of a better name for `f` and `g`?

File: src/bishnu_poudel_ex/ex01/tidy_code.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+
+# user input guess
+def input_guess():
+    return int(input('Your guess: '))
+
+
+# random number between 1 and 12
```
Comment (huynhngoc) [inline src/bishnu_poudel_ex/ex01/tidy_code.py]: The result of this function cannot between 1 and 12. It's rolling 2 dices, and the result is from 2 to 12.

File: src/bishnu_poudel_ex/ex01/comp_to_loop.py (hunk 1 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/comp_to_loop.py]: Done. removed the part.

File: src/bishnu_poudel_ex/ex01/deck_by_comp.py (hunk 2 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/deck_by_comp.py]: modified to compact form.

File: src/bishnu_poudel_ex/ex01/deck_by_comp.py (hunk 3 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/deck_by_comp.py]: Done

File: src/bishnu_poudel_ex/ex01/letter_counts.py (hunk 4 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/letter_counts.py]: Done.

File: src/bishnu_poudel_ex/ex01/letter_counts.py (hunk 5 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/letter_counts.py]: done.

File: src/bishnu_poudel_ex/ex01/tidy_code.py (hunk 6 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/tidy_code.py]: removed the as part. also renamed many other variables

File: src/bishnu_poudel_ex/ex01/tidy_code.py (hunk 8 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/tidy_code.py]: done

File: src/bishnu_poudel_ex/ex01/tidy_code.py (hunk 9 above)
Comment (vsnupoudel) [reply src/bishnu_poudel_ex/ex01/tidy_code.py]: modified. done.

**Pull request merged**